src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/AttachedCollections/BrowseObjectBase.cs

```diff
@@ -0,0 +1,60 @@
+﻿// Licensed to the .NET Foundation under one or more agreements. The .NET Foundation licenses this file to you under the MIT license. See the LICENSE.md file in the project root for more information.
+
+using System;
+using System.ComponentModel;
+using Microsoft.VisualStudio.Shell;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Tree.Dependencies.AttachedCollections
+{
+    /// <summary>
+    /// This is a slightly modified copy of Microsoft.VisualStudio.Shell.LocalizableProperties.
+    /// http://index/#Microsoft.VisualStudio.Shell.12.0/LocalizableProperties.cs.html
+    /// Unfortunately we can't reuse that class because the GetComponentName method on
+    /// it is not virtual, so we can't provide a name string for the VS Property Grid's
```

Comment:
Why don't we just make it virtual? That would be preferable versus adding public API to our assembly for something that doesn't belong in this assembly.

Reply:
https://dev.azure.com/devdiv/DevDiv/_git/VS/pullrequest/248337